blogposts/inside-rust/01-announcement.md

```diff
@@ -0,0 +1,166 @@
+    reserve the freedom to change this
+* If you enable `-Cpanic=abort`, we are able to optimize the size of binaries
+  to remove most code related to unwinding.
+  * In extreme cases, it should be possible to remove **all** traces of
+    unwinding, when you know that it will not occur.
+  * In practice, most "C" functions are never expected to unwind (because they
+    are written in C, for example, and not in C++).
+    * However, because unwinding is now part of most system ABIs, even C
+      functions can unwind -- most notably, `pthread_cancel` can cause all
+      manner of C functions to unwind, including common functions like `read`
+      and `write`.  (For a complete list, search for "cancellation points" in
+      the [pthreads man
+      page](http://man7.org/linux/man-pages/man7/pthreads.7.html).)
+* Changing the behavior from `-Cpanic=unwind` to `-Cpanic=abort` should not
+  cause Undefined Behavior.
+  * However, this may not be tenable, or at least not without making binaries
+    much larger. See the discussion below for more details.
+  * It may, of course, cause programs to abort that used to execute
+    successfully. This could occur if a panic would've been caught and
+    recovered.
+
+## The primary question: should the `"C"` ABI permit unwinding?
+
+The core question that we would like to decide is whether the `"C"` ABI, as
+defined by Rust, should permit unwinding.
+
+This is not a question we expected to be debating. We've long declared that
+unwinding through Rust's `"C"` ABI is undefined behavior. In part, this is
+because nobody had spent the time to figure out what the correct behavior would
+be, or how to implement it, although (as we'll see shortly) there are other
+good reasons for this choice. 
+
+In any case, in PR #65646, @Amanieu proposed that we could, in fact, simply
+define the behavior of unwinding across `"C"` boundaries. In discussing this,
+discovered that the question of whether the `"C"` ABI should permit unwinding was
+less clear-cut than we had assumed.
+
+If the `"C"` ABI does not permit unwinding, a new ABI, called `"C unwind"`,
+will be introduced specifically to support unwinding.
+
+## Three specific proposals
+
+The project group has narrowed the design space down to three specific
+proposals. Two of these introduce the new `"C unwind"` ABI, and one does not.
+
+Each proposal specifies the behavior of each type of unwind (Rust `panic!`,
+foreign (e.g. C++), and forced (e.g. `pthread_exit`)) when it encounters an
+ABI boundary under either the `panic=unwind` or `panic=abort` compile-mode.
+
+### 2 ABIs, minimal specification
+
+* `"C"` ABI boundary, `panic=<any>`
+  * `panic!`: program aborts
+  * forced unwind, no destructors: unwind behaves normally
+  * all other foreign unwinding: undefined behavior
+* `"C unwind"` ABI boundary
+  * With `panic=unwind`: all types of unwinding behave normally
+  * With `panic=abort`: all types of unwinding abort the program
```

Comment:
Hmm, actually, I think this entry strays somewhat from what we had in the doc, right? The "minimal spec" there avoided distinguishing forced unwinding from other kinds of exceptions, which simplified life, but it meant that we couldn't insert helpful shims to catch unwinding and abort, because we couldn't locally tell when unwinding might occur. 

In any case, this setup as described here means that one cannot `longjmp` over a "C unwind" boundary in panic=abort mode, which seems not great. This is why I think we carved out a special case in the paper doc which made "forced unwinding with destructors" UB *even in panic=unwind*.

Reply:
This is option 1c described in the doc. Option 1 is the one below which carves out the special exception for forced unwinding.